app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/agent/SelfEmployedCYAController.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.agent
+
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import uk.gov.hmrc.http.{HeaderCarrier, InternalServerException, NotFoundException}
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.SelfEmploymentDataKeys.businessAccountingMethodKey
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.config.AppConfig
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.config.featureswitch.FeatureSwitch.SaveAndRetrieve
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.config.featureswitch.FeatureSwitching
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.IncomeTaxSubscriptionConnector
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.utils.ReferenceRetrieval
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.models.{AccountingMethodModel, SelfEmploymentsCYAModel}
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.services.{AuthService, MultipleSelfEmploymentsService}
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.views.html.agent.SelfEmployedCYA
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class SelfEmployedCYAController @Inject()(val checkYourAnswersView: SelfEmployedCYA,
+                                          val authService: AuthService,
+                                          val incomeTaxSubscriptionConnector: IncomeTaxSubscriptionConnector,
+                                          multipleSelfEmploymentsService: MultipleSelfEmploymentsService,
+                                          mcc: MessagesControllerComponents)
+                                         (implicit val appConfig: AppConfig, val ec: ExecutionContext)
+  extends FrontendController(mcc) with I18nSupport with FeatureSwitching with ReferenceRetrieval {
+
+
+  def show(id: String, isEditMode: Boolean): Action[AnyContent] = Action.async { implicit request =>
+    if (isEnabled(SaveAndRetrieve)) {
+      withReference { reference =>
```

Comment:
The `authService.authorised()` is missing, check out `app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/agent/BusinessAccountingMethodController.scala` to have an example.
It's also missing on the individual `BusinessAccountingMethodController`.

Reply:
Do you mean the individual SelfEmployedCYAController?  I've just added it there, anyway, but both BusinessAccountingMethodControllers look correct to me.

Reply:
Sorry, yes, I meant `SelfEmployedCYAController`

Reply:
Done, good spot.